src/sentry/utils/sentry_apps/webhooks.py

```diff
@@ -36,6 +40,58 @@ def wrapper(sentry_app, app_platform_event, url=None):
     return wrapper
 
 
+def check_broken(sentryapp: SentryApp, org_id: str):
+    redis_key = sentryapp._get_redis_key(org_id)
+    buffer = IntegrationRequestBuffer(redis_key)
+    if buffer.is_integration_broken():
+        org = Organization.objects.get(id=org_id)
+        if features.has("organizations:disable-sentryapps-on-broken", org):
+            sentryapp._disable()
+            notify_disable(org, sentryapp.slug, redis_key)
+            buffer.clear()
+
+        extra = {
+            "sentryapp_webhook": sentryapp.webhook_url,
+            "sentryapp_slug": sentryapp.slug,
+            "sentryapp_uuid": sentryapp.uuid,
+            "org_id": org_id,
+            "buffer_record": buffer._get_all_from_buffer(),
+        }
+        logger.info(
+            "sentryapp.disabled",
+            extra=extra,
+        )
+
+
+def record_timeout(sentryapp: SentryApp, org_id: str, e: Union[ConnectionError, Timeout]):
+    """
+    Record Unpublished Sentry App timeout or connection error in integration buffer to check if it is broken and should be disabled
+    """
+    if sentryapp.is_published:
```

Comment:
Maybe instead of `is_published` we could check `if not sentryapp.is_internal` so we're not potentially disabling ones that are in the to-be-public pipeline. @scefali have we ever had incident worthy errors from an unpublished app? Like while developing they cause so many errors it's a problem?

Reply:
@ceorourke Yea, i think that makes sense